### PR TITLE
Feature/make http request timeout configurable

### PR DESCRIPTION
**Pull request by nick6655**
I read over the contributors guide and am hoping I understand correctly that for minor changes like this I can propose the feature in the form of a pull request which would implement it.

The feature I am proposing is to expose the timeout via the Configuration class in a similar fashion to skip_ssl_verification. This will help to alleviate an issue I am experiencing with latency between data centers causing timeout errors. The ability to configure this value I believe should be universally helpful.

### CODE REVIEW

**Review by pyashish — APPROVED** [review]
Looks really great! Thanks Nick.

**Comment by nick6655** [comment]
Thanks @pyashish I noticed the workflow which I assume is GitHub runners or something of the like still needs to be approved a run. Not sure if that's something the maintainers of the project typically do or something that once it's approved I need to initiate. Please let me know at your earliest convenience. Also could you please give a rough ETA for the merge to be completed. We will need this as soon as we can get it.

**Pull request merged**
